### PR TITLE
Merged Dockerfile.access-om*, all models reference models.json

### LINKED ISSUE #64: Store model names in one location and re-use
We're hard-coding the model names in a couple of places. It would be good to use a global env variable to store the model names to avoid having them defined in two locations:

See https://stackoverflow.com/a/73274812

e.g.
```yaml
env:
  models: ("access-om2" "access-om3")
```
and then use them in the options and later in the generate-matrix job.

This was brought up in this PR:

https://github.com/ACCESS-NRI/build-ci/pull/62#discussion_r1300879895

@CodeGat replied

> Unfortunately, the 'choice' type doesn't yet allow dynamic choices yet... https://github.com/orgs/community/discussions/12029
only other way would be to have them input the model name or 'all' manually as a string...Not sure if thats the best idea either, but it'd be better than a static choice I suppose.

Making this issue so when https://github.com/orgs/community/discussions/12029 is implemented this can be done.

### PR DESCRIPTION
**Pull request by CodeGat**
* Changed build-and-push-image-build workflow input to `string` rather than `choice`, added extra error checking. 
* Merged Dockerfile.access-om* into Dockerfile.dependency
* Pulled out hardcoded references to models, now uses `containers/models.json`

Should close #64 !

### CODE REVIEW

**Review by jo-basevi — COMMENTED** [review]
Just a couple of  small changes - otherwise looks good to me

File: .github/workflows/dependency-image-build.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+jobs:
+  setup-dependency-image:
+    runs-on: ubuntu-latest
+    outputs: 
+      model-components: ${{ steps.get-model-components.outputs.model-components }}
+    steps:
+      - name: Get model components for ${{ inputs.model }}
+        id: get-model-components
+        run: echo "model-components=$(jq -cr 'keys' containers/models.json)" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (CodeGat) [reply .github/workflows/dependency-image-build.yml]: Thanks for picking that up!

File: .github/workflows/dependency-image-build.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    needs:
+      - setup-dependency-image
+    uses: access-nri/build-ci/.github/workflows/build-and-push-image.yml@main
+    with:
+      container-registry: ghcr.io
+      container-name: access-nri/build-${{ inputs.model}}-${{ inputs.compiler-name }}${{ inputs.compiler-version }}-${{ inputs.spack-packages-version }}
+      dockerfile-directory: containers
+      dockerfile-name: Dockerfile.dependency
+      build-args: |
+        "PACKAGES=${{ needs.setup-dependency-image.outputs.model-components }}"
```
[the remark this answers is not in the document]
Comment (CodeGat) [reply .github/workflows/dependency-image-build.yml]: Correct - will update

**Review by CodeGat — COMMENTED** [review]
Implemented recommendations

**Review by jo-basevi — APPROVED** [review]
Thanks, all looks good to me!

File: .github/workflows/build-and-push-image-build.yml [hunk 3]
```diff
@@ -30,9 +27,14 @@ jobs:
         id: get-model
         run: |
           if [[ "${{ inputs.model }}" = "all" ]]; then
-            echo "models=['access-om2', 'access-om3']" >> $GITHUB_OUTPUT
+            echo "models=$(jq -cr 'keys' containers/models.json)" >> $GITHUB_OUTPUT
```
Comment (aidanheerdegen) [inline .github/workflows/build-and-push-image-build.yml:30]: 👀

Comment (CodeGat) [reply .github/workflows/build-and-push-image-build.yml:30]: much more maintainable, too :D

**Pull request merged**